module/zfs/spa_misc.c

```diff
@@ -2582,6 +2582,15 @@ spa_set_missing_tvds(spa_t *spa, uint64_t missing)
 const char *
 spa_state_to_name(spa_t *spa)
 {
+	ASSERT3P(spa, !=, NULL);
+
+	/*
+	 * it is possible for the spa to exist, without root vdev
+	 * as the spa transitions during import/export
+	 */
+	if (spa->spa_root_vdev == NULL) {
+		return ("TRANSITIONING");
+	}
```

Comment:
Thanks.  I think it would be a good idea to go one small step farther and store the `spa_root_vdev` in a local variable.  That still won't entirely eliminate this hard to hit race (since this is done outside the lock intentionally), but it should prevent the NULL deref.

```c
vdev_t *rvd = spa->spa_root_vdev;
if (rvd == NULL)
        ...

vdev_state_t state = rcv->vdev_state;
...
````

Reply:
I agree, will do